client.go

```diff
@@ -0,0 +1,318 @@
+	// http client errors
+	clientErrBuildRequest = "could not build http request"
+	clientErrSendRequest  = "could not send http request"
+	// http response payload errors
+	clientErrReadResponse = "could not read response body"
+	clientErrNon200       = "non 200 status code received"
+	clientErrUnmarshal    = "could not unmarshal response body"
+)
+
+// NewClient is the default constructor for the client
+func NewClient(url, token string, httpClient *http.Client) (*Client, error) {
+	if url == "" {
+		return nil, errors.New(clientErrURLEmpty)
+	}
+	if token == "" {
+		return nil, errors.New(clientErrAuthEmpty)
+	}
+	if httpClient == nil {
+		httpClient = http.DefaultClient
+	}
+	return &Client{
+		hostURL:    url,
+		authStr:    token,
+		httpClient: httpClient,
+	}, nil
+}
+
+func (c *Client) addAuth(r *http.Request) {
+	r.Header.Set("Authorization", c.authStr)
+}
+
+// Dump retrieves all reputation entries
+func (c *Client) Dump() ([]Reputation, error) {
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/dump", c.hostURL), nil)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %s", clientErrBuildRequest, err)
+	}
+	c.addAuth(req)
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %s", clientErrSendRequest, err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("%s: %d", clientErrNon200, resp.StatusCode)
+	}
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil {
+		return nil, fmt.Errorf("%s: %s", clientErrReadResponse, err)
+	}
+	var ret []Reputation
+	if err = json.Unmarshal(bodyBytes, &ret); err != nil {
+		return nil, fmt.Errorf("%s: %s", clientErrUnmarshal, err)
+	}
+	return ret, nil
+}
+
+// Heartbeat checks whether an IPrepd deployment is healthy / reachable
+func (c *Client) Heartbeat() (bool, error) {
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/__heartbeat__", c.hostURL), nil)
+	if err != nil {
+		return false, fmt.Errorf("%s: %s", clientErrBuildRequest, err)
+	}
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return false, fmt.Errorf("%s: %s", clientErrSendRequest, err)
+	}
+	return (resp.StatusCode == http.StatusOK), nil
+}
+
+// LBHeartbeat checks whether an IPrepd LB is healthy / reachable
+func (c *Client) LBHeartbeat() (bool, error) {
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/__lbheartbeat__", c.hostURL), nil)
+	if err != nil {
+		return false, fmt.Errorf("%s: %s", clientErrBuildRequest, err)
+	}
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return false, fmt.Errorf("%s: %s", clientErrSendRequest, err)
+	}
+	return (resp.StatusCode == http.StatusOK), nil
+}
+
+// GetReputation fetches the reputation of a given object and type
+func (c *Client) GetReputation(objectType, object string) (*Reputation, error) {
+	if object == "" {
+		return nil, errors.New(clientErrObjectEmpty)
+	}
+	if objectType == "" {
+		return nil, errors.New(clientErrObjectTypeEmpty)
+	}
```

Comment:
is it worth calling the functions from `validators.go` here and anywhere else we publish data to verify the type and object?

Reply:
✅